tools/global_bss_check.py

```diff
@@ -0,0 +1,216 @@
+        exit(1)
+
+    buildMap = parseMapFile(mapFileBuild)
+    expectedMap = parseMapFile(mapFileExpected)
+
+    comparedDict = collections.OrderedDict()
+
+    for symbol in buildMap:
+        if symbol in expectedMap:
+            comparedDict[symbol] = Compared( buildMap[symbol].vram, buildMap[symbol].file, expectedMap[symbol].vram, expectedMap[symbol].file, buildMap[symbol].vram - expectedMap[symbol].vram )
+            if comparedDict[symbol].diff != 0:
+                badFiles.add(buildMap[symbol].file)
+                
+        else:
+            missingFiles.add(buildMap[symbol].file)
+            comparedDict[symbol] = Compared( buildMap[symbol].vram, buildMap[symbol].file, -1, "", "Unknown" )
+
+    for symbol in expectedMap:
+        if not symbol in buildMap:
+            missingFiles.add(expectedMap[symbol].file)
+            comparedDict[symbol] = Compared( -1, "", expectedMap[symbol].vram, expectedMap[symbol].file, "Unknown" )
+
+    return badFiles, missingFiles, comparedDict
+
+
+def printCsv(badFiles, missingFiles, comparedDict, printAll = True):
+    print("Symbol Name,Build Address,Build File,Expected Address,Expected File,Difference,GOOD/BAD/MISSING")
+
+    allGood = True
+    allFound = True
+    
+    problemFiles = set.union(badFiles, missingFiles)
+
+    # If it's bad or missing, don't need to do anything special.
+    # If it's good, check for if it's in a file with bad or missing stuff, and check if print all is on. If none of these, print it.
+    
+    for symbol in comparedDict:
+        symbolInfo = comparedDict[symbol]
+        symbolGood = colorama.Fore.RED + "BAD" + colorama.Fore.RESET
+        if type(symbolInfo.diff) != int:
+            symbolGood = colorama.Fore.YELLOW + "MISSING" + colorama.Fore.RESET
+            print(f"{symbol},{symbolInfo.buildAddress:X},{symbolInfo.buildFile},{symbolInfo.expectedAddress:X},{symbolInfo.expectedFile},{symbolInfo.diff},{symbolGood}")
+            continue
+
+        if symbolInfo.diff == 0:
+            symbolGood = colorama.Fore.GREEN + "GOOD" + colorama.Fore.RESET
+            if (not symbolInfo.buildFile in badFiles and not symbolInfo.expectedFile in badFiles) and (not symbolInfo.buildFile in badFiles and not symbolInfo.expectedFile in badFiles) and not printAll:
+                continue
+        
+        if symbolInfo.buildFile != symbolInfo.expectedFile:
+            symbolGood += colorama.Fore.CYAN + " MOVED" + colorama.Fore.RESET
+        print(f"{symbol},{symbolInfo.buildAddress:X},{symbolInfo.buildFile},{symbolInfo.expectedAddress:X},{symbolInfo.expectedFile},{symbolInfo.diff:X},{symbolGood}")
+
+
+def main():
+    description = "Check that globally visible bss has not been reordered."
+    # TODO
+    epilog = """\
+    N.B. Since this script reads the map files, it can only see globally visible bss; in-function static bss must be examined with other tools.
+    """
```

Comment:
I think this `TODO` isn't needed anymore, or is it?
```suggestion
    description = "Check that globally visible bss has not been reordered."
    epilog = """\
    N.B. Since this script reads the map files, it can only see globally visible bss; in-function static bss must be examined with other tools.
    """
```